syntax = "proto3";
package presence.v1;

service PresenceService {
  rpc Heartbeat(HeartbeatReq) returns (Ack);
  rpc Roster(RoomId) returns (RosterReply);
  rpc SubscribePresence(RoomId) returns (stream PresenceEvent);
}

message RoomId { string room_id = 1; }

message HeartbeatReq {
  string room_id = 1;
  string user_id = 2;
  string display_name = 3; 
}

message Ack { bool ok = 1; string error = 2; }

message PresenceUser {
  string user_id = 1;
  string display_name = 2;   
  int64 last_seen_ms = 3;
}

message RosterReply {
  repeated PresenceUser users = 1; // CHANGED
}

enum PresenceType { JOIN = 0; LEAVE = 1; }
message PresenceEvent {
  PresenceType type = 1;
  string room_id = 2;
  string user_id = 3;
  string display_name = 4;   
  int64 ts_ms = 5;
}
